configure.ac

```diff
@@ -866,6 +866,10 @@ AC_SUBST([EXPORT_MACOSX_DEPLOYMENT_TARGET])
 CONFIGURE_MACOSX_DEPLOYMENT_TARGET=
 EXPORT_MACOSX_DEPLOYMENT_TARGET='#'
 
+# Record the value of IOS_DEPLOYMENT_TARGET enforced by the selected host triple.
+CONFIGURE_IOS_DEPLOYMENT_TARGET=${IOS_DEPLOYMENT_TARGET}
+AC_SUBST([CONFIGURE_IOS_DEPLOYMENT_TARGET])
```

Comment:
It's an isolation against the local environment. If there's an environment variable named IOS_DEPLOYMENT_TARGET, the Makefile will pick that up, and that will bake the local environment value into `sysconfig`. The iOS config reads the requested iOS minim target from the triple (e.g., `arm64-apple-ios12.0`), and then passes that into compiler arguments as `-mios-version-min=12.0`; we want to ensure that the `sysconfig` version bakes *this* value in, not the local environment variable version.

Reply:
The macOS config does a similar thing, for the same reason.

Reply:
> If there's an environment variable named IOS_DEPLOYMENT_TARGET, the Makefile will pick that up, and that will bake the local environment value into `sysconfig`

The [Make manual](https://www.gnu.org/software/make/manual/html_node/Environment.html) says:
* an explicit assignment in the makefile overrides the environment
* variables that came from the make’s environment are placed into the environment of the commands

So if IOS_DEPLOYMENT_TARGET is in the environment when running `make`, it will be overridden by the Makefile, and the overridden value passed to the commands. If IOS_DEPLOYMENT_TARGET wasn't in the environment when running `make`, then it won't be passed to the commands at all. 

It looks like the sysconfigdata file is actually generated by `generate_posix_vars` in `sysconfig/__main__.py`, which parses the Makefile for lines that look like variable assignments, and doesn't use the environment at all.

On macOS the configure script sets an EXPORT_MACOSX_DEPLOYMENT_TARGET variable, which is set either to "#" or "" to control whether an `export MACOSX_DEPLOYMENT_TARGET` line in the Makefile is commented out. That's how we control the compiler's target platform version on macOS – there is no version argument added to CFLAGS. If I understand correctly, the equivalent for iOS would be IPHONEOS_DEPLOYMENT_TARGET (not IOS_...). 

However, that means that every package build system needs to pull MACOSX_DEPLOYMENT_TARGET out of the sysconfigdata and set that environment variable manually, e.g. [here in setuptools](https://github.com/pypa/setuptools/blob/v69.2.0/setuptools/_distutils/util.py#L75). So I agree that using CFLAGS for iOS is better, because then all the build systems will pick it up automatically.

In summary, I think the current implementation is fine, even if it doesn't work in the way you expected. My only suggestion is to rename the variable so it doesn't resemble the official Apple ones, which could confuse people into thinking that it controls the compiler directly. Maybe IOS_VERSION_MIN, to match the command line argument?

Reply:
> So if IOS_DEPLOYMENT_TARGET is in the environment when running `make`, it will be overridden by the Makefile, and the overridden value passed to the commands. If IOS_DEPLOYMENT_TARGET wasn't in the environment when running `make`, then it won't be passed to the commands at all.

You're right - I think in my testing around this, I wasn't differentiating between "Defined, but empty" and "not defined", so I was seeing environment variables leak in.

> If I understand correctly, the equivalent for iOS would be IPHONEOS_DEPLOYMENT_TARGET (not IOS_...).

Good catch - the variable should definitely be `IPHONEOS_DEPLOYMENT_TARGET`; I'll adjust the code accordingly.

> In summary, I think the current implementation is fine, even if it doesn't work in the way you expected. My only suggestion is to rename the variable so it doesn't resemble the official Apple ones, which could confuse people into thinking that it controls the compiler directly. Maybe IOS_VERSION_MIN, to match the command line argument?

I'm not sure about this one. My original naming goof notwithstanding, `IPHONEOS_DEPLOYMENT_TARGET` is a known value in the iOS ecosystem; inventing a new name means we need to define what that variable means, describe how (if at all) it interacts with `IPHONEOS_DEPLOYMENT_TARGET` etc (even if the answer is "it doesn't"). 

To the best of my knowledge, explicit command line arguments take precedence over environment variables in Apple toolchains; so someone with `IPHONEOS_DEPLOYMENT_TARGET` in their environment would also see the `--mios-version-min` flags, and would be able to conclude that the explicit flag was taking priority. 

With that said - this is definitely a "strong opinion, weakly held"; if the core team feels we should use a separate variable to avoid confusion, I'll gladly update to reflect that.

Reply:
That makes sense, but I think it's worth updating the comment in the Makefile to explain this. Something like:
```
# Unlike on macOS, the iOS deployment target is controlled using `-mios-version-min`
# arguments added to CFLAGS by the configure script. So this variable is not used during 
# the build, and is only listed here to add it to the sysconfigdata.